.github/workflows/orchestrion.yml

```diff
@@ -0,0 +1,21 @@
+name: Orchestrion
+on:
+  workflow_dispatch: # manually
+  schedule: # nightly
+    - cron: "0 0 * * *"
+  pull_request:
+  merge_group:
+  push:
+    branches:
+      - release-v*
+
+concurrency:
+  # Automatically cancel previous runs if a new one is triggered to conserve resources.
+  group: ${{ github.workflow }}-${{ github.event_name }}-${{ github.ref }}
+
+jobs:
+  test:
+    name: 'Run Tests'
+    uses: DataDog/orchestrion/.github/workflows/workflow_call.yml@eliott.bouhana/smoke-tests
+    with:
+      dd-trace-go-ref: ${{ github.event_name == 'pull_request' && github.event.pull_request.head.sha || github.sha }}
```

Comment:
Can we not use `github.sha` in all situations? It feels like it would be best to test using the "merge candidate" instead of the PR `HEAD`; although I'm not sure whether that commit is actually visible for `go get` to work in the target workflow (in which case, feel free to ignore my ramblings).